MekHQ/src/mekhq/campaign/mission/AtBDynamicScenarioFactory.java

```diff
@@ -786,67 +724,380 @@ public static int generateForce(AtBDynamicScenario scenario, AtBContract contrac
             currentLanceWeightString = currentLanceWeightString.substring(1);
         }
 
-        // If over budget for both BV and unit count, pull units until it works
+        // If over budget for BV or unit count, pull units until it works
         while (forceUnitBudget > 0 && generatedEntities.size() > forceUnitBudget) {
-            generatedEntities.remove(Compute.randomInt(generatedEntities.size()));
+            int targetUnit = Compute.randomInt(generatedEntities.size());
+            generatedEntities.remove(targetUnit);
         }
 
         if (forceTemplate.getGenerationMethod() == ForceGenerationMethod.BVScaled.ordinal()) {
-            while (((forceBV / forceBVBudget * 100) > targetPercentage) && (generatedEntities.size() > 1)) {
-                int targetEntity = Compute.randomInt(generatedEntities.size());
-                forceBV -= generatedEntities.get(targetEntity).calculateBattleValue();
-                generatedEntities.remove(targetEntity);
+            String balancingType = "";
+            if (campaign.getCampaignOptions().isUseGenericBattleValue()) {
+                balancingType = " Generic";
             }
+            logger.info("Generated a force with " + forceBV + '/' + forceBVBudget + ' ' + balancingType + " BV");
```

Comment:
I know the Info method does support variable args and would be better to use the string formatting stuff.

Reply:
It never occurred to me to use String.format, thanks for the suggestion

Reply:
Yea, it's a vague security issue that ensures the data you're trying to pass into the string is actually what you intended to pass into it.

IE: Not trying to pass an object (for some reason) to a string when it is expecting a number.

Reply:
Sounds good. Won't take me long to get that resolved tomorrow. It'll be a useful thing for future too.